### ISSUE TITLE
Update broke our NuGetPack process

### ISSUE THREAD

**Issue by jamescrowley** [fsharp/FAKE#643]
Can anyone explain the purpose of this commit:

https://github.com/fsharp/FAKE/commit/2ddc4d4c57693f974cd49feeb0d2adcd3d631d2f

? It's broken our build script when we just updated our FAKE version. 

We previously did:

        Target "Package" (fun _ ->
            !! (currentDirectory @@ "db" @@ "Database.nuspec")
            |>  Seq.iter (NuGetPack (fun param -> { param with 
                                                    ToolPath = nuGetPath
                                                    OutputPath = artifactsDir
                                                    WorkingDir = currentDirectory
                                                    IncludeReferencedProjects = true
                                                    Version = buildVersion
                                                    Properties = [("Configuration", buildMode)] }))
        )

but now Database.nuspec is written to the root of our project (currentDirectory), and then packages up our entire source code tree. Previously it just ran nuget pack on the nuspec file in the db directory and packaged that up. 

I'd fix this but I don't understand the new behavior caused by 
https://github.com/fsharp/FAKE/commit/2ddc4d4c57693f974cd49feeb0d2adcd3d631d2f enough - why is it always writing (and deleting) a nuspec file even if you're giving it one that exists?

**Comment by forki** [comment]
I think you need to set the working dir to currentDirectory @@ "db"

**Comment by forki** [comment]
@ilkerde do you remember what changed?

**Comment by jamescrowley** [comment]
@forki that's what I've been trying as a fix, but this then breaks subsequent items: 

    Target "Package" (fun _ ->
        !! (currentDirectory @@ "db" @@ "Database.nuspec")
        ++ (currentDirectory @@ "rules" @@ "Rules.nuspec")
        ++ (currentDirectory @@ "src" @@ "TestApi" @@ "TestApi.csproj")
        |>  Seq.iter (fun path -> 
            path |> NuGetPack (fun param -> { param with 
                                                ToolPath = nuGetPath
                                                OutputPath = artifactsDir
                                                WorkingDir = Path.GetDirectoryName(path)
                                                IncludeReferencedProjects = true
                                                Version = buildVersion
                                                Properties = [("Configuration", buildMode)] }))
    )

**Comment by jamescrowley** [comment]
The db nuget package gets created okay with the above fix, but I then get 

       Creating .nuspec file at src\TestApi\TestApi.csproj0.0.0.0-beta.nuspec
       Created nuspec file src\TestApi\TestApi.csproj0.0.0.0-beta.nuspec

which I haven't figured out yet

**Comment by forki** [comment]
> why is it always writing (and deleting) a nuspec file even if you're giving it one that exists?

IIRC because we want to replace placeholders.

**Comment by jamescrowley** [comment]
I think this last bit is because it's copied the .csproj file as if it was a nuspec file?

**Comment by jamescrowley** [comment]
Pretty sure this is a bug now. We're relying on NuGet pack behaviour such that you can pass in a .csproj file, and have a .nuspec on the filesystem alongside it, and nuget will combine the two. Pretty sure this templating (even ignoring the bad copying of the .csproj) would break this.

**Comment by forki** [comment]
I never knew we supported this scenario ;-)
Can you come up with a fix which enables both?

**Comment by ilkerde** [comment]
@forki  Thanks for letting me know about this. Maybe I can add some value to this discussion. 

@jamescrowley I think my pull request to this commit is descriptive in terms of why I did this. (see: https://github.com/fsharp/FAKE/pull/554). The reason why it is creating and deleting the (intermediate) .nuspec file is because `NugetPack` supports transforming a nuspec template now. I chose to copy this "strategy" of nuspec preparation from the original `Nuget` [implementation](https://github.com/fsharp/FAKE/blob/fd7334ef5ebc07ab4f2f8ae57babc8eca7910924/src/app/FakeLib/NuGet/NugetHelper.fs#L348). That is, this create/delete thing was kind of always there for `Nuget`. I just introduced it for `NugetPack` as well. Aside, I (somewhat) kept backwards compatibility since the original version expected a plain (ready to go) .nuspec file only, without any placeholders in there. That's basically the story.

Hope it helps at least.
